comp/core/workloadmeta/store_test.go

```diff
@@ -679,6 +679,49 @@ func TestGetKubernetesDeployment(t *testing.T) {
 	tassert.True(t, errors.IsNotFound(err))
 }
 
+func TestGetKubernetesNamespace(t *testing.T) {
+	deps := fxutil.Test[dependencies](t, fx.Options(
+		logimpl.MockModule(),
+		config.MockModule(),
+		fx.Supply(NewParams()),
+	))
+
+	s := newWorkloadmetaObject(deps)
+
+	namespace := &KubernetesNamespace{
+		EntityID: EntityID{
+			Kind: KindKubernetesNamespace,
+			ID:   "default",
+		},
+	}
+
+	s.handleEvents([]CollectorEvent{
+		{
+			Type:   EventTypeSet,
+			Source: fooSource,
+			Entity: namespace,
+		},
+	})
+
+	retrievedNamespace, err := s.GetKubernetesNamespace("default")
+	tassert.NoError(t, err)
+
+	if !reflect.DeepEqual(namespace, retrievedNamespace) {
+		t.Errorf("expected namespace %q to match the one in the store", retrievedNamespace.ID)
+	}
+
+	s.handleEvents([]CollectorEvent{
+		{
+			Type:   EventTypeUnset,
+			Source: fooSource,
+			Entity: namespace,
+		},
+	})
```

Comment:
Question

After this we are asserting that the `datadog-cluster-agent` namespace doesn't exist. 
So why are we handling the same event again here? Is this necessary?

Reply:
good point, removed!